### ISSUE TITLE
Version 1.5.0 do not work algorithms 

### ISSUE THREAD

**Issue by EvgeniyKorepov** [nicehash/excavator#225]
algorithms do not work (from what I checked) CryptoNightV7 and daggerhashimoto_decred
Config:
[
   {"time":1,"commands":[
      {"id":1,"method":"subscribe","params":["nhmp.eu.nicehash.com:3200","3DQ9fRMVfxHaTkxt2y8BBDnVb1eAad94NP.home"]}
   ]},
   {"time":2,"commands":[
      {"id":1,"method":"algorithm.add","params":["cryptonightV7"]}
  ]},
   {"time":3,"commands":[
      {"id":1,"method":"worker.add","params":["cryptonightV7","0"]}
   ]},
   {"time":10,"loop":10,"commands":[
      {"id":1,"method":"worker.print.speed","params":["0"]},
      {"id":1,"method":"algorithm.print.speeds","params":[]}
   ]}
]

Result:
[13:41:37][0x00003b58][info] Log started
[13:41:37][0x00003b58][info] core | Found CUDA device: GeForce GTX 960
[13:41:37][0x00003b58][info] api | Listening on 0.0.0.0:33333
[13:41:37][0x00003b58][info] http | Listening on 0.0.0.0:38080
[13:41:37][0x00003b58][info] core | Initialized!
[13:41:38][0x00003b58][info] Connected to nhmp.eu.nicehash.com:3200
[13:41:38][0x00000ab8][info] nhmp | Subscribed
[13:41:39][0x00003b58][info] nhmp | cryptonightV7 | add failure
[13:41:39][0x00003b58][info] nhmp | Reason: Internal server error (-1)
[13:41:40][0x00001224][info] wrkr0-0 | Algorithm: CUDA-cryptonightV7 parameters: B=14,TPB=32
[13:41:43][0x00003b58][info] nhmp | Disconnected
[13:41:43][0x00003b58][warning] nhmp | read: WebSocket connection closed normally

daggerhashimoto_decred :

[15:08:51][0x000009f8][info] Log started
[15:08:51][0x000009f8][info] core | Found CUDA device: GeForce GTX 960
[15:08:51][0x000009f8][info] api | Listening on 0.0.0.0:33333
[15:08:51][0x000009f8][info] http | Listening on 0.0.0.0:38080
[15:08:51][0x000009f8][info] core | Initialized!
[15:08:53][0x000009f8][info] Connected to nhmp.eu.nicehash.com:3200
[15:08:53][0x00004644][info] nhmp | Subscribed
[15:08:55][0x00003314][info] wrkr0-0 | Algorithm: CUDA-daggerhashimoto_decred parameters: R_0:R_1=3:1,P=0
[15:08:57][0x000009f8][info] nhmp | Disconnected
[15:08:57][0x000009f8][warning] nhmp | read: WebSocket connection closed normally
[15:09:01][0x00004644][info] core | Device #0-0 speed: 0.000000 H/s
[15:09:07][0x00004644][info] Connected to nhmp.eu.nicehash.com:3200
[15:09:07][0x000009f8][info] nhmp | Subscribed
[15:09:11][0x00004644][info] nhmp | Disconnected
[15:09:11][0x00004644][warning] nhmp | read: WebSocket connection closed normally
[15:09:11][0x000009f8][info] core | Device #0-0 speed: 0.000000 H/s
[15:09:21][0x000009f8][info] Connected to nhmp.eu.nicehash.com:3200
[15:09:21][0x00004644][info] nhmp | Subscribed
[15:09:21][0x000009f8][info] core | Device #0-0 speed: 0.000000 H/s
[15:09:25][0x00004644][info] nhmp | Disconnected
[15:09:25][0x00004644][warning] nhmp | read: WebSocket connection closed normally
[15:09:31][0x000009f8][info] core | Device #0-0 speed: 0.000000 H/s
[15:09:35][0x000009f8][info] Connected to nhmp.eu.nicehash.com:3200
[15:09:35][0x00004644][info] nhmp | Subscribed
[15:09:39][0x000009f8][info] nhmp | Disconnected
[15:09:39][0x000009f8][warning] nhmp | read: WebSocket connection closed normally
[15:09:41][0x00004644][info] core | Device #0-0 speed: 0.000000 H/s

**Comment by dropky** [comment]
CryptoNightV7 is not supported on the stratums (yet), monero fork is scheduled on 6th April, untill then the algorithm is useless (stratums will be updated in the next couple of hours though). Other algorithms should work. What is your config file for dual mining? You have to add both algorithm separately, like this:

```
[
     {"time":1,"commands":[
       {"id":1,"method":"subscribe","params":["nhmp.eu.nicehash.com:3200","btc"]}
    ]},
    {"time":2,"commands":[
        {"id":1,"method":"algorithm.add","params":["daggerhashimoto"]},
	{"id":1,"method":"algorithm.add","params":["decred"]}
    ]},
    {"time":3,"commands":[
        {"id":1,"method":"worker.add","params":["daggerhashimoto_decred", "3"]}
    ]},
    {"time":10,"loop":10,"commands":[
        {"id":1,"method":"worker.print.speeds","params":[]}
    ]}
]  
```

**Comment by EvgeniyKorepov** [comment]
Indeed, there was a wrong configuration file. I made it following the logic, but I did not suspect that now it is not there. :-(

The new version of API is definitely not an April Fool's joke?

Now, to get the list of algorithms, I should use the method algorithm.list, not the expected method, but the worker.list method. After that, check all the workers, and check all the algorithms in the workers to find out what algorithms are running on the Excavator?

**Comment by dropky** [comment]
Sorry, no. We had to change the API in order to support new stratum protocol. Here are some changes in the logic behind the API: 

- subscribe method establish connection to the stratum server (there is only one connection opened at all times, even if multiple algorithms are running);
- with algorithm.add method miner "subscribes" to the specific jobs; That is way you have to add both daggerhashimoto and decred when running dual mining; When you call algorithm.add stratum will start sending jobs to miner;
- handling with workers is the same as before; the only difference is that you have to assign worker to algorithm name not algorithm id;

Algorithm.list method will return all the algorithms that are running and the combined speeds (similar to algorithm.print.speeds method). If you only want to find out what algorithms are running on excavator you can use the algorithm.list method or worker.list method (both will provide you that information). If you want to get the information for speed & algorithm per device/worker you'll have to call worker.list method. I don't know what kind of information do you need, my guess is you'll probably have to call worker.list method. You can get all the information from there (take a look how we modified the index.html code).

**Comment by MichaelMiner** [comment]
CryptoNightV7 is not supported on the stratums (yet),
![v7](https://user-images.githubusercontent.com/33788644/38006419-9b6098fc-324d-11e8-988f-5797fbbcbb73.jpg)
